test/e2e/suite/conformance/certificates/tests.go

```diff
@@ -795,7 +795,7 @@ func (s *Suite) Define() {
 			By("Sanity-check the issued Certificate")
 			err = f.Helper().ValidateCertificate(f.Namespace.Name, "testcert", validations...)
 			Expect(err).NotTo(HaveOccurred())
-		}, featureset.OnlySAN)
```

Comment:
Ah, I see here that 'LongDomain' is defined as something that contains a subdomain segment that is `maxLengthOfDomainSegment` long (which I think is 63 characters) - I don't think any public ACME servers/Let's Encrypt's staging environment has a restriction on this? 🤔 if it does, and the 'pebble' based ACME server does not, then Pebble needs modifying to also fail in these cases as it aims to replicate the ACME RFC as closely as possible

Reply:
The issue is with Let's Encrypt specifically rather than the ACME protocol. There's a discussion here: https://github.com/certbot/certbot/issues/1915

Let's encrypt copies one of the names from the subject alternative name (which supports arbitrary length domains) to the Common Name field (which is useless in modern TLS). Unfortunately, this field has a maximum length of 64 `char`s (C definition, so bytes).

This isn't a problem with cert-manager as we don't set the subject at all